packages/financial-templates-lib/src/clients/FinancialContractEventClient.ts

```diff
@@ -174,65 +216,63 @@ class FinancialContractEventClient {
       this.contractType == "ExpiringMultiParty" // If the contract is an EMP then find the SettleExpiredPosition events.
         ? this.financialContract.getPastEvents("SettleExpiredPosition", blockSearchConfig)
         : this.financialContract.getPastEvents("SettleEmergencyShutdown", blockSearchConfig), // Else, find the SettleEmergencyShutdown events.
-    ]);
+    ] as any[])) as [
+      string,
+      LiquidationCreatedEvent[],
+      DisputeEvent[],
+      DisputeSettlementEvent[],
+      CreateEvent[],
+      NewSponsorEvent[],
+      DepositEvent[],
+      WithdrawEvent[],
+      RedeemEvent[],
+      RegularFeeEvent[],
+      FinalFeeEvent[],
+      LiquidationWithadrawnEvent[],
+      SettleExpiredEvent[] | SettleEmergencyShutdownEvent[]
+    ];
     // Set the current contract time as the last update timestamp from the contract.
-    this.lastUpdateTimestamp = currentTime;
+    this.lastUpdateTimestamp = parseInt(currentTime);
 
     // Process the responses into clean objects.
     // Liquidation events.
-    for (let event of liquidationEventsObj) {
+    for (const event of liquidationEventsObj) {
       this.liquidationEvents.push({
         transactionHash: event.transactionHash,
         blockNumber: event.blockNumber,
-        sponsor: event.returnValues.sponsor,
-        liquidator: event.returnValues.liquidator,
-        liquidationId: event.returnValues.liquidationId,
-        tokensOutstanding: event.returnValues.tokensOutstanding,
-        lockedCollateral: event.returnValues.lockedCollateral,
-        liquidatedCollateral: event.returnValues.liquidatedCollateral,
+        ...event.returnValues,
       });
     }
 
     // Dispute events.
-    for (let event of disputeEventsObj) {
+    for (const event of disputeEventsObj) {
```

Comment:
nice. changes like this, while not ts specific are great to include for general code quality.

Reply:
For whatever reason, eslint/the typescript compiler _only_ complain about this after changing the filename to `ts` haha, which is why I changed them.